File: android/app/src/main/java/com/nxtend/team35/yubiboard/vision/MarkerDetectionResult.kt
package com.nxtend.team35.yubiboard.vision

data class NormalizedPoint(
    val x: Float,
    val y: Float,
)

data class DetectedMarker(
    val id: Int,
    val center: NormalizedPoint,
    val corners: List<NormalizedPoint>,
)

data class MarkerDetectionResult(
    val capturedAtMonotonicMs: Long,
    val sourceWidth: Int,
    val sourceHeight: Int,
    val markers: List<DetectedMarker>,
    val stable: Boolean,
    val stableFrameCount: Int = 0,
    val requiredStableFrames: Int = MarkerStabilityTracker.DEFAULT_REQUIRED_STABLE_FRAMES,
    /** マーカー座標が表示（ミラー補正済み）座標系か。前面カメラも補正するため常に true。 */
    val mirrorCorrected: Boolean = true,
    val cameraFacing: String = "back",
)

File: ios/Screact/Vision/MarkerDetectionResult.swift
import Foundation

struct NormalizedPoint: Equatable {
    let x: Float
    let y: Float
}

struct DetectedMarker: Equatable {
    let id: Int
    let center: NormalizedPoint
    let corners: [NormalizedPoint]
}

struct MarkerDetectionResult {
    let capturedAtMonotonicMs: Int64
    let sourceWidth: Int
    let sourceHeight: Int
    let markers: [DetectedMarker]
    let stable: Bool
    let stableFrameCount: Int
    let requiredStableFrames: Int

    init(capturedAtMonotonicMs: Int64,
         sourceWidth: Int,
         sourceHeight: Int,
         markers: [DetectedMarker],
         stable: Bool,
         stableFrameCount: Int = 0,
         requiredStableFrames: Int = MarkerStabilityTracker.defaultRequiredStableFrames) {
        self.capturedAtMonotonicMs = capturedAtMonotonicMs
        self.sourceWidth = sourceWidth
        self.sourceHeight = sourceHeight
        self.markers = markers
        self.stable = stable
        self.stableFrameCount = stableFrameCount
        self.requiredStableFrames = requiredStableFrames
    }
}
